appcelerator/titanium_mobile
# refactor(ios): optimize promise creation on iOS 12

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28412

**Summary:**
- Avoids doing a JS `eval()` every time a promise is returned by a Titanium API on iOS 12.
- Theoretically a small memory leak would occur every time a promise is created on iOS 12 every time a JS function is defined via `eval()`. _(Not proven.)_

**Test:**
1. Build and run the below code on iOS 12.
2. Tap on the "Create Array Buffer" button.
3. Verify alert dialog displays: "Created Buffer 1"
4. Tap on the "Create Array Buffer" button again.
5. Verify alert dialog displays: "Created Buffer 2"

```javascript
let count = 0;
const window = Ti.UI.createWindow();
const button = Ti.UI.createButton({
	title: "Create Array Buffer",
});
button.addEventListener("click", async () => {
	count++;
	const tiBuffer = Ti.createBuffer({ value: `Created Buffer ${count}`, type: Ti.Codec.ASCII });
	const arrayBuffer = await tiBuffer.toBlob().arrayBuffer();
	const byteArray = new Uint8Array(arrayBuffer);
	let message = "";
	for (nextByte of byteArray) {
		message += String.fromCharCode(nextByte);
	}
	alert(message);
});
window.add(button);
window.open();
```
